pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -54,24 +55,30 @@ def _process_conn_spec(conn_spec):
         raise TypeError("conn_spec must be a string or dict")
 
 
-def _process_syn_spec(syn_spec, conn_spec, prelength, postlength, connect_np_arrays):
+def _process_syn_spec(syn_spec, conn_spec, prelength, postlength, use_connect_arrays):
     """Processes the synapse specifications from None, string or dictionary to a dictionary."""
     if syn_spec is None:
+        # for use_connect_arrays, return "static_synapse" by default
+        if use_connect_arrays:
+            return {"synapse_model": "static_synapse"}
```

Comment:
This does not simplify because of the ``return syn_spec`` below, does it?
It would require an additional ``elif syn_spec is None`` below...
Would it be clearer if I changed ``return syn_spec`` into ``return None``?

Reply:
@Silmathoron You are right, I had misunderstood the code. Would it then not be clearer to have an explicit `else` clause:
```python
if use_connect_arrays:
    return {"synapse_model": "static_synapse"}
else:
   return syn_spec
```
?

Reply:
I don't think so, the return-early pattern is widely advised as it avoid potential errors with forgotten if/else clauses and reduces the amount of code.
I would really recommend leaving it that way.